Address safer CPP warnings in WKPagePrivateMac.mm

csaavedra opened the pull request:
#### 54b554d727ccc2a9255b638a0c6bededc7e0b242
<pre>
Address safer CPP warnings in WKPagePrivateMac.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=300984">https://bugs.webkit.org/show_bug.cgi?id=300984</a>

Reviewed by Ryosuke Niwa.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
(-[WKObservablePageState initWithPage:]):
(-[WKObservablePageState _protectedPage]):
(-[WKObservablePageState dealloc]):
(-[WKObservablePageState isLoading]):
(-[WKObservablePageState title]):
(-[WKObservablePageState URL]):
(-[WKObservablePageState hasOnlySecureContent]):
(-[WKObservablePageState _webProcessIsResponsive]):
(-[WKObservablePageState estimatedProgress]):
(-[WKObservablePageState unreachableURL]):
(-[WKObservablePageState serverTrust]):
(WKPageGetObjectRegistry):
(WKPageIsURLKnownHSTSHost):
(WKPageLoadURLRequestReturningNavigation):
(WKPageLoadFileReturningNavigation):
(WKPageGetWebView):
(WKPageIsPlayingVideoInEnhancedFullscreen):
(WKPageGetAccessibilityWebProcessDebugInfo):
(WKPageAccessibilityClearIsolatedTree):

Canonical link: <a href="https://commits.webkit.org/301732@main">https://commits.webkit.org/301732@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/696c759b896fd6d030686a52d6f6e15c12200c82

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126844 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46485 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37462 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133848 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78450 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47108 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55018 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96529 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129792 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37696 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113459 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77043 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36569 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77242 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107525 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31941 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136374 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53514 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41195 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105038 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54010 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109818 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104734 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26716 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50242 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28582 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50984 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53443 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59294 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52697 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56032 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54446 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

csaavedra commented on Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
```diff
@@ -55,56 +55,62 @@ - (id)initWithPage:(RefPtr<WebKit::WebPageProxy>&&)page
     return self;
 }
 
+- (Ref<WebKit::WebPageProxy>)_protectedPage
+{
+    return *_page;
+}
+
```
I'm now wondering if one could just mark `_page` as `const` and lazy-initialize it instead.

rniwa replied:
If we can do that, that might be preferable.

csaavedra committed:
Address safer CPP warnings in WKPagePrivateMac.mm
https://bugs.webkit.org/show_bug.cgi?id=300984

Reviewed by Ryosuke Niwa.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
(-[WKObservablePageState initWithPage:]):
(-[WKObservablePageState _protectedPage]):
(-[WKObservablePageState dealloc]):
(-[WKObservablePageState isLoading]):
(-[WKObservablePageState title]):
(-[WKObservablePageState URL]):
(-[WKObservablePageState hasOnlySecureContent]):
(-[WKObservablePageState _webProcessIsResponsive]):
(-[WKObservablePageState estimatedProgress]):
(-[WKObservablePageState unreachableURL]):
(-[WKObservablePageState serverTrust]):
(WKPageGetObjectRegistry):
(WKPageIsURLKnownHSTSHost):
(WKPageLoadURLRequestReturningNavigation):
(WKPageLoadFileReturningNavigation):
(WKPageGetWebView):
(WKPageIsPlayingVideoInEnhancedFullscreen):
(WKPageGetAccessibilityWebProcessDebugInfo):
(WKPageAccessibilityClearIsolatedTree):

Canonical link: https://commits.webkit.org/301732@main

cdumez commented on Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
```diff
@@ -55,61 +55,67 @@ - (id)initWithPage:(RefPtr<WebKit::WebPageProxy>&&)page
     return self;
 }
 
+- (Ref<WebKit::WebPageProxy>)_protectedPage
```
@darinadler mentioned in earlier reviews that he prefers us using a static C++ function rather than a ObjC method like this (Please see WKPreferences.mm as example).

csaavedra replied:
Will make a note to update this and other classes, there are probably many other places where we have used ObjC methods too.

darinadler replied:
The reasons I suggested functions were:
1) They can be private to the source file when they are used only in one file, typically the implementation file of the class in question.
2) They have less runtime overhead than a method since there is no method dispatch and are eligible to be inlined by the compiler.

If those reasons don’t apply, we could still use an Objective-C property or method.
